src/modules/socket.js

```diff
@@ -1,12 +1,186 @@
+      authorId: chat.authorId._id,
+      authorName: chat.authorId.nickname,
+      authorProfileUrl: chat.authorId.profileImageUrl,
+      content: chat.content,
+      time: chat.time,
+      isValid: chat.isValid,
+      inOutNames: chat.inOutNames,
+    });
+  }
+  return chatsToSend;
+};
+
+// FCM 알림으로 보내는 content는 채팅 type에 따라 달라집니다.
+// type이 text인 경우 `${nickname}: ${content}`를, 아닌 경우 `${nickname}`를 보냅니다.
+const getMessageBody = (type, nickname, content) => {
+  // TODO: 채팅 메시지 유형에 따라 Body를 다르게 표시합니다.
+  if (type === "text") {
+    // 채팅 메시지 유형이 텍스트인 경우 본문은 "${nickname}: ${content}"가 됩니다.
+    return `${nickname}: ${content}`;
+  } else if (type === "s3img") {
+    // 채팅 유형이 이미지인 경우 본문은 "${nickname} 님이 이미지를 전송하였습니다"가 됩니다.
+    // TODO: 사용자 언어를 가져올 수 있으면 개선할 수 있다고 생각합니다.
+    const suffix = " 님이 이미지를 전송하였습니다.";
+    return `${nickname} ${suffix}`;
+  } else if (type === "in" || type === "out") {
+    // 채팅 메시지 type이 "in"이거나 "out"인 경우 본문은 "${nickname} 님이 입장하였습니다" 또는 "${nickname} 님이 퇴장하였습니다"가 됩니다.
+    // TODO: 사용자 언어를 가져올 수 있으면 개선할 수 있다고 생각합니다.
+    const suffix =
+      type === "in" ? " 님이 입장하였습니다" : "님이 퇴장하였습니다";
+    return `${nickname} ${suffix}`;
+  } else if (type === "payment" || type === "settlement") {
+    // 채팅 메시지 type이 "in"이거나 "out"인 경우 본문은 "${nickname} 님이 결제를 완료하였습니다" 또는 "${nickname} 님이 정산을 완료하였습니다"가 됩니다.
+    // TODO: 사용자 언어를 가져올 수 있다면 개선할 수 있다고 생각합니다.
+    const suffix =
+      type === "payment"
+        ? " 님이 결제를 완료하였습니다"
+        : " 님이 정산을 완료하였습니다";
+    return `${nickname} ${suffix}`;
+  }
+};
+
+/**
+ * 채팅을 전송하고 채팅 알림을 발생시킵니다.
+ * @summary express 라우터에서 채팅 이벤트를 보낼 수 있게 함수를 분리했습니다.
+ * @param {Server<DefaultEventsMap, DefaultEventsMap, DefaultEventsMap, any>} io - Socket.io 서버 인스턴스입니다. req.app.get("io")를 통해 접근할 수 있습니다.
+ * @param {Object} chat - 채팅 메시지 내용입니다.
+ * @param {string} chat.roomId - 채팅 및 채팅 알림을 보낼 방의 ObjectId입니다.
+ * @param {string} chat.type - 채팅 메시지의 유형입니다. "text" | "s3img" | "in" | "out" | "payment" | "settlement" 입니다.
+ * @param {string} chat.content - 채팅 메시지의 본문입니다. chat.type이 "s3img"인 경우에는 채팅의 objectId입니다. chat.type이 "in"이거나 "out"인 경우 입퇴장한 사용자의 id(!==ObjectId)입니다.
+ * @param {string} chat.authorId - 채팅을 보낸 사용자의 ObjectId입니다.
+ * @param {Date?} chat.time - optional. 채팅 메시지 전송 시각입니다.
+ * @return {Promise<Boolean>} 채팅 및 알림 전송에 성공하면 true, 중간에 오류가 발생하면 false를 반환합니다.
+ */
+const emitChatEvent = async (io, chat) => {
+  try {
+    // chat must contain type, content and authorId
+    // chat can contain time or not.
+    if (
+      !io ||
+      !chat?.roomId ||
+      !chat?.type ||
+      !chat?.content ||
+      !chat?.authorId
+    ) {
+      throw new IllegalArgumentsException();
+    }
+
+    const { roomId, type, content, authorId } = chat;
+    const time = chat?.time || Date.now();
+    const { nickname, profileImageUrl } = await userModel.findById(
+      authorId,
+      "nickname profileImageUrl"
+    );
+    if (!nickname) {
+      throw new IllegalArgumentsException();
+    }
```

Comment:
```suggestion
    const { nickname, profileImageUrl } = await userModel.findById(
      authorId,
      "nickname profileImageUrl"
    );
    const { name, part } = await roomModel.findById(roomId, "name part");
    if (!nickname || !profileImageUrl || !name || !part) {
      throw new IllegalArgumentsException();
    }
```

아래 room 검색 코드를 위로 올리고 검색 결과 확인도 하면 좋을 것 같아요